### PR TITLE
Fixing bug in the Date::parseDate() method

### PR DESCRIPTION
**Pull request by lorenzo**
When using a different timezone it would return different dates
because of the timezone changes.

### CODE REVIEW

File: src/Database/Type/DateType.php [hunk 1]
```diff
@@ -99,20 +99,4 @@ protected function _parseValue($value)
...
-    public function testToImmutableAndToMutable()
-    {
-        $this->type->useImmutable();
-        $this->assertInstanceOf('DateTimeImmutable', $this->type->marshal('2015-11-01'));
-        $this->assertInstanceOf('DateTimeImmutable', $this->type->toPhp('2015-11-01', $this->driver));
-
-        $this->type->useMutable();
-        $this->assertInstanceOf('DateTime', $this->type->marshal('2015-11-01'));
-        $this->assertInstanceOf('DateTime', $this->type->toPhp('2015-11-01', $this->driver));
-    }
```
Comment (lorenzo) [inline src/Database/Type/DateType.php]: This code was placed in a non-test class :D

Comment (markstory) [inline src/Database/Type/DateType.php]: I'm so good at coding sometimes.

**Comment by markstory** [comment]
Travis is failing in new and entertaining ways :cry:

File: src/I18n/DateFormatTrait.php [hunk 2]
```diff
@@ -55,6 +57,13 @@
     protected static $_jsonEncodeFormat = "yyyy-MM-dd'T'HH:mm:ssZ";
 
     /**
+     * Caches whehter or not this class is a subclass of a Date or MutableDate
```
Comment (antograssiot) [inline src/I18n/DateFormatTrait.php]: typo whether

File: src/Database/Type/DateType.php (hunk 1 above)
Comment (HavokInspiration) [inline src/Database/Type/DateType.php]: Looks like we are good at reviewing it too.

**Comment by lorenzo** [comment]
@markstory sigh :(

**Comment by markstory** [comment]
Nice work @lorenzo I never even considered that we had all those open cursors.

**Pull request merged**